pkg/integration/components/test_driver.go

```diff
@@ -0,0 +1,236 @@
+package components
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/jesseduffield/lazygit/pkg/config"
+	"github.com/jesseduffield/lazygit/pkg/gui/types"
+	integrationTypes "github.com/jesseduffield/lazygit/pkg/integration/types"
+)
+
+type TestDriver struct {
+	gui          integrationTypes.GuiDriver
+	keys         config.KeybindingConfig
+	pushKeyDelay int
+	*assertionHelper
+	shell *Shell
+}
+
+func NewTestDriver(gui integrationTypes.GuiDriver, shell *Shell, keys config.KeybindingConfig, pushKeyDelay int) *TestDriver {
+	return &TestDriver{
+		gui:             gui,
+		keys:            keys,
+		pushKeyDelay:    pushKeyDelay,
+		assertionHelper: &assertionHelper{gui: gui},
+		shell:           shell,
+	}
+}
+
+// key is something like 'w' or '<space>'. It's best not to pass a direct value,
+// but instead to go through the default user config to get a more meaningful key name
+func (self *TestDriver) press(keyStr string) {
+	self.Wait(self.pushKeyDelay)
+
+	self.gui.PressKey(keyStr)
+}
+
+func (self *TestDriver) typeContent(content string) {
+	for _, char := range content {
+		self.press(string(char))
+	}
+}
+
+func (self *TestDriver) ContinueMerge() {
```

Comment:
not sure whether to wrap these misc helpers into their own struct. Will wait and see.